hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,377 @@
+        1. we are using the REST API CLI
+        2. we have to go to every vlan and then remove our port from it;
+           that makes multiple calls.
+        """
+        for vlan in self._get_vlans(interface):
+            self._remove_vlan_from_trunk(interface, vlan[1])
+
+    def _set_native_vlan(self, interface, vlan):
+        """ Set the native vlan of an interface.
+
+        Args:
+            interface: interface to set the native vlan to
+            vlan: vlan to set as the native vlan
+
+        Method relies on the REST API CLI which is slow
+        """
+        if not self._is_port_on(interface):
+            self._port_on(interface)
+        command = 'interface vlan ' + vlan + '\r\n untagged ' + \
+            self.interface_type + ' ' + interface
+        self._execute(interface, CONFIG, command)
+
+    def _remove_native_vlan(self, interface):
+        """ Remove the native vlan from an interface.
+
+        Args:
+            interface: interface to remove the native vlan from.vlan
+        """
+        vlan = self._get_native_vlan(interface)[1]
+        command = 'interface vlan ' + vlan + '\r\n no untagged ' + \
+            self.interface_type + ' ' + interface
+        self._execute(interface, CONFIG, command)
+
+    def _port_shutdown(self, interface):
+        """ Shuts down <interface>
+
+        Turn off portmode hybrid, disable switchport, and then shut down the
+        port. All non-default vlans must be removed before calling this.
+        """
+
+        url = self._construct_url(interface=interface)
+        interface = self._convert_interface(self.interface_type) + \
+            interface.replace('/', '-')
+        payload = '<interface><name>%s</name><portmode><hybrid>false' \
+                  '</hybrid></portmode><shutdown>true</shutdown>' \
+                  '</interface>' % interface
+
+        self._make_request('PUT', url, data=payload)
+
+    def _port_on(self, interface):
+        """ Turns on <interface>
+
+        Turn on port and enable hybrid portmode and switchport.
+        """
+
+        url = self._construct_url(interface=interface)
+        interface = self._convert_interface(self.interface_type) + \
+            interface.replace('/', '-')
+        payload = '<interface><name>%s</name><portmode><hybrid>true' \
+                  '</hybrid></portmode><switchport></switchport>' \
+                  '<shutdown>false</shutdown></interface>' % interface
+
+        self._make_request('PUT', url, data=payload)
+
+    def _is_port_on(self, port):
+        """ Returns a boolean that tells the status of a switchport"""
+
+        # the url here requires a suffix to GET the shutdown tag in response.
+        url = self._construct_url(interface=port) + '\?with-defaults'
+        response = self._make_request('GET', url)
+        root = etree.fromstring(response.text)
+        shutdown = root.find(self._construct_tag('shutdown')).text
+        # is error handling needed here?
```

Comment:
Probably for robustness, we should do something like:

```python
if shutdown not in ('true', 'false'):
    raise SomeKindOfError()
return shutdown == 'false'
```

...rather than treating possibly bogus data as the port being off, which could result in various information leaks.